### PR TITLE
ROX-24169: support OCM service accounts

### PR DESCRIPTION
**Pull request by stehessel**
### Description

OCM has deprecated the use of OCM offline tokens for authentication. We must therefore add support for service account authentication to cloud source integrations. The plan is to remove OCM token support completely in future releases.

**Note: This PR is currently blocked by this OCM bug https://issues.redhat.com/browse/OCM-7933. Until this is resolved, service accounts cannot carry the needed permissions and the e2e test is expected to fail with 403s. Other than that, the PR is ready for review.**

### User-facing documentation

- [x] CHANGELOG is updated **OR** update is not needed
- [x] [documentation PR](https://spaces.redhat.com/display/StackRox/Submitting+a+User+Documentation+Pull+Request) is created and is linked above **OR** is not needed

### Testing and quality

- [x] the change is production ready: the change is GA or otherwise the functionality is gated by a feature flag
- [x] CI results are inspected

#### Automated testing

- [ ] added unit tests
- [ ] added e2e tests
- [ ] added regression tests
- [ ] added compatibility tests
- [x] modified existing tests

#### How I validated my change

see e2e test

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -15,7 +15,9 @@ Please avoid adding duplicate information across this changelog and JIRA/doc inp
 
 ### Removed Features
 
-### Deprecated Fatures
+### Deprecated Features
+
+- ROX-24169: API token authentication has been deprecated by Red Hat OpenShift Cluster Manager. The corresponding cloud source integration now uses service accounts for authentication.
```
Comment (janisz) [inline CHANGELOG.md:20]: Can we link to their announcement?

File: central/cloudsources/datastore/datastore_impl.go [hunk 2]
```diff
@@ -97,7 +97,8 @@ func validateCloudSource(cloudSource *storage.CloudSource) error {
 	if cloudSource.GetName() == "" {
 		errorList.AddString("cloud source name must be defined")
 	}
-	if cloudSource.GetCredentials().GetSecret() == "" {
+	if cloudSource.GetCredentials().GetSecret() == "" &&
```
Comment (janisz) [inline central/cloudsources/datastore/datastore_impl.go:100]: Should we have a todo to remove it
Also should we show log with information about deprecated feature used?

File: central/cloudsources/manager/manager.go [hunk 3]
```diff
@@ -64,7 +64,8 @@ type managerImpl struct {
 }
 
 func newManager(cloudSourcesDS cloudSourcesDS.DataStore,
-	discoveredClustersDS discoveredClustersDS.DataStore, clustersDS clusterDS.DataStore) *managerImpl {
+	discoveredClustersDS discoveredClustersDS.DataStore, clustersDS clusterDS.DataStore,
+) *managerImpl {
```
Comment (janisz) [inline central/cloudsources/manager/manager.go:68]:
```suggestion
	discoveredClustersDS discoveredClustersDS.DataStore, clustersDS clusterDS.DataStore) *managerImpl {
```

**Review by janisz — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (stehessel) [reply CHANGELOG.md:20]: Sadly their is no public announcement as per the OCM team.

File: central/cloudsources/manager/manager.go [hunk 4]
```diff
@@ -143,7 +144,10 @@ func (m *managerImpl) getDiscoveredClustersFromCloudSources() []*discoveredclust
 		return nil
 	}
 
-	clients, err := createClients(cloudSources)
+	createTimeout := 30 * time.Second
```
Comment (dashrews78) [inline central/cloudsources/manager/manager.go:147]: nit:  would it make sense to make this a constant.

File: pkg/cloudsources/ocm/client.go [hunk 5]
```diff
@@ -31,7 +31,8 @@ func NewClient(config *storage.CloudSource, options ...opts.ClientOpts) (*ocmCli
 	connection, err := sdkClient.NewConnectionBuilder().
 		RetryLimit(opt.Retries).
 		URL(urlfmt.FormatURL(config.GetOcm().GetEndpoint(), urlfmt.HTTPS, urlfmt.NoTrailingSlash)).
-		Tokens(config.GetCredentials().GetSecret()).Agent(clientconn.GetUserAgent()).Build()
+		Client(config.GetCredentials().GetClientId(), config.GetCredentials().GetClientSecret()).
+		Tokens(config.GetCredentials().GetSecret()).Agent(clientconn.GetUserAgent()).BuildContext(ctx)
```
Comment (dashrews78) [inline pkg/cloudsources/ocm/client.go:35]: nit:  Formatting here looks off

**Review by dashrews78 — APPROVED** [review]
couple nits.  Otherwise LG

**Pull request merged**